### PR TITLE
Add support for memory hotplug virt-install options

### PR DESCRIPTION
**Pull request by sathnaga**
Add support for memory hotplug virt-install options

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/libvirt_vm.py [hunk 1]
```diff
@@ -459,7 +459,8 @@ def add_hvm_or_pv(help_text, hvm_or_pv):
                 logging.warning("Unknown virt type hvm_or_pv, using default.")
                 return ""
 
-        def add_mem(help_text, mem, maxmem=None, hugepage=False):
+        def add_mem(help_text, mem, maxmem=None, hugepage=False,
+                    hotplugmaxmem=None, hotplugmemslots=1):
```
Comment (balamuruhans) [inline virttest/libvirt_vm.py]: why do we have to set `hotplugmemslot` default as 1 ?, it can be `None`

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -474,6 +475,17 @@ def add_mem(help_text, mem, maxmem=None, hugepage=False):
...
+                    if not has_sub_option('memory', 'hotplugmemorymax'):
+                        logging.warning("hotplugmemorymax option not supported"
+                                        "by virt-install")
+                    else:
+                        cmd += ",hotplugmemorymax=%s" % hotplugmaxmem
+                    if not has_sub_option('memory', 'hotplugmemoryslots'):
+                        logging.warning("hotplugmemoryslots option not "
+                                        "supported by virt-install")
+                    else:
+                        cmd += ",hotplugmemoryslots=%d" % int(hotplugmemslots)
```
Comment (balamuruhans) [inline virttest/libvirt_vm.py]: type cast is not necessary as you have already taken care of it during assignment.

File: virttest/libvirt_vm.py (hunk 1 above)
Comment (sathnaga) [reply virttest/libvirt_vm.py]: `hotplugmaxmem` and `hotplugmemslots` are related, just to make sure the default value incase user not set `hotplugmemslots` explicitly, not sure of default slot value, let me check and set it accordingly.

File: virttest/libvirt_vm.py (hunk 2 above)
Comment (sathnaga) [reply virttest/libvirt_vm.py]: sure, will update it.

File: virttest/libvirt_vm.py (hunk 1 above)
Comment (sathnaga) [reply virttest/libvirt_vm.py]: @balamuruhans it expects both, so default as `1` in safest option for now, ```
Starting install...
ERROR    XML error: both maximum memory size and memory slot count must be specified
Domain installation does not appear to have been successful.
```

**Comment by sathnaga** [comment]
@balamuruhans addressed the comments

**Pull request merged**